core-rust/core-api-server/src/core_api/handlers/lts/stream_transactions_basic_outcomes.rs

```diff
@@ -0,0 +1,136 @@
+use crate::core_api::*;
+use radix_engine::transaction::BalanceChange;
+use state_manager::{
+    jni::state_manager::ActualStateManager,
+    store::traits::{QueryableProofStore, QueryableTransactionStore},
+    CommittedTransactionIdentifiers, LedgerTransactionOutcome, LedgerTransactionReceipt,
+};
+
+#[tracing::instrument(skip(state), err(Debug))]
+pub(crate) async fn handle_rc_stream_transactions_basic_outcomes(
+    state: State<CoreApiState>,
+    request: Json<models::LtsStreamTransactionsBasicOutcomesRequest>,
+) -> Result<Json<models::LtsStreamTransactionsBasicOutcomesResponse>, ResponseError<()>> {
+    core_api_read_handler(
+        state,
+        request,
+        handle_rc_stream_transactions_basic_outcomes_internal,
+    )
+}
+
+fn handle_rc_stream_transactions_basic_outcomes_internal(
+    state_manager: &ActualStateManager,
+    request: models::LtsStreamTransactionsBasicOutcomesRequest,
+) -> Result<models::LtsStreamTransactionsBasicOutcomesResponse, ResponseError<()>> {
+    assert_matching_network(&request.network, &state_manager.network)?;
+
+    let from_state_version: u64 = extract_api_state_version(request.from_state_version)
+        .map_err(|err| err.into_response_error("from_state_version"))?;
+
+    let limit: u64 = request
+        .limit
+        .try_into()
+        .map_err(|_| client_error("limit cannot be negative"))?;
+
+    if limit == 0 {
+        return Err(client_error("limit must be positive"));
+    }
+
+    if limit > MAX_STREAM_COUNT_PER_REQUEST.into() {
+        return Err(client_error(format!(
+            "limit must <= {MAX_STREAM_COUNT_PER_REQUEST}"
+        )));
+    }
+
+    let max_state_version = state_manager.store().max_state_version();
+
+    let txns = state_manager.store().get_committed_transaction_bundles(
+        from_state_version,
+        limit.try_into().expect("limit out of usize bounds"),
+    );
+
+    let mapping_context = MappingContext::new(&state_manager.network);
+
+    let api_txns = txns
+        .into_iter()
+        .map(|(_ledger_transaction, receipt, identifiers)| {
+            Ok(to_api_lts_comitted_transaction_basic_outcome(
+                &mapping_context,
+                receipt,
+                identifiers,
+            )?)
+        })
+        .collect::<Result<Vec<models::LtsCommittedTransactionFungibleOutcome>, ResponseError<()>>>(
+        )?;
+
+    let start_state_version = if api_txns.is_empty() {
+        0
```

Comment:
This logic feels... wrong?  But I know it was copied from the previous place.
I feel it should just be returning from_state_version - I'm not sure why this check makes sense. If you agree, can you fix it in all the places? 👍 :)